### ISSUE TITLE
Using Oscar with Bottom-up features 

### ISSUE THREAD

**Issue by l3bjorge** [Alcoholrithm/Oscar_Scripts#2]
Hi thanks for this amazing work!

I've tried to use your code to generate captions on raw images with bottom-up features but I'm getting bad results. I moved the notebook to a python script and tried inference on 1 image with the model 'OursL(XE)' you posted on your MODEL ZOO. I'm printing the features and classes detected and they seem to be fine but the caption generated has nothing to do with the image.

This is the image i'm using:

![perro_frisbee](https://user-images.githubusercontent.com/25321261/146541134-bf969e44-6213-4736-85aa-c871174b1733.jpg)

This is the result I get after running the code.

python test.py
Config 'workspace/detectron2/configs/VG-Detection/faster_rcnn_R_101_C4_caffe.yaml' has no VERSION. Assuming it to be compatible with latest v2.
Modifications for VG in RPN (modeling/proposal_generator/rpn.py):
	Use hidden dim 512 instead fo the same dim as Res4 (1024).

Modifications for VG in RoI heads (modeling/roi_heads/roi_heads.py):
	1. Change the stride of conv1 and shortcut in Res5.Block1 from 2 to 1.
	2. Modifying all conv2 with (padding: 1 --> 2) and (dilation: 1 --> 2).
	For more details, please check 'https://github.com/peteanderson80/bottom-up-attention/blob/master/models/vg/ResNet-101/faster_rcnn_end2end_final/test.prototxt'.

/home/jhurtado/anaconda3/envs/Alcoholrithm/lib/python3.7/site-packages/torch/functional.py:445: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  ../aten/src/ATen/native/TensorShape.cpp:2157.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
[{'class': 'dog'}, {'class': 'frisbee'}, {'class': 'leg'}, {'class': 'leg'}, {'class': 'head'}, {'class': 'shadow'}, {'class': 'eye'}, {'class': 'eyes'}, {'class': 'beach'}, {'class': 'snow'}, {'class': 'dog'}, {'class': 'ear'}, {'class': 'eye'}, {'class': 'paw'}, {'class': 'legs'}, {'class': 'water'}, {'class': 'nose'}, {'class': 'ground'}, {'class': 'ground'}, {'class': 'water'}, {'class': 'shadow'}, {'class': 'ground'}, {'class': 'snow'}, {'class': 'ground'}, {'class': 'snow'}, {'class': 'head'}, {'class': 'mouth'}, {'class': 'sand'}]
[[array([[0.36432788, 0.23231444, 0.        , ..., 0.99799365, 0.31319785,
        0.9006105 ],
       [0.        , 0.01907516, 0.        , ..., 0.337672  , 0.15087502,
        0.07700826],
       [0.        , 0.        , 0.        , ..., 0.9307157 , 0.08133312,
        0.28387186],
       ...,
       [0.02281303, 0.04437782, 0.0105911 , ..., 0.37423557, 0.34131297,
        0.36085242],
       [0.        , 0.00111979, 0.        , ..., 0.31540072, 0.06328667,
        0.08102156],
       [0.        , 0.93097776, 0.01253132, ..., 0.99827766, 0.82969713,
        0.52183276]], dtype=float32), array([117, 123, 808, 808, 191, 683, 467, 546,  62, 176, 117, 274, 467,
       786, 829, 183, 391, 465, 465, 183, 683, 465, 176, 465, 176, 191,
       452, 326])]]
/home/jhurtado/image_captioning/Pruebas/MaravillasDeSergio/Oscar_Scripts/Oscar/oscar/modeling/modeling_utils.py:506: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  beam_id = idx // vocab_size
['a bunch of tools sitting on top of a pile of pens and pencils.']

I'm thinking there might be a missmatch between labels and features but I don't how can I try to fix this. 
Do you think you can help me with this?

Thanks a lot in advance

**Comment by Alcoholrithm** [comment]
The decoder of Oscar generates image caption based on feature vectors and labels of the image.
However, each encoder creates different feature vectors and labels.
That's why, you should always use the appropriate encoder for each decoder.

The encoder of Ours(XE) is Oscar repo's VINVL model. So it is natural that strange caption came out.
Fortunately, I have encoder that was trained on bottom up feature.
If you leave an email address in the comment, I'll send you the model as soon as possible.

I got ['a dog with the frisbee in its mouth on a beach.' ] from above image using my bottom up model.
If you have any problem please leave a comment at any time.

Thank you :)

**Comment by Alcoholrithm** [comment]
The decoder of Oscar generates image caption based on feature vectors and labels of the image.
However, each encoder creates different feature vectors and labels.
That's why, you should always use the appropriate encoder for each decoder.

The encoder of Ours(XE) is Oscar repo's VINVL model. So it is natural that strange caption came out.
Fortunately, I have encoder that was trained on bottom up feature.
If you leave an email address in the comment, I'll send you the model as soon as possible.

I got ['a dog with the frisbee in its mouth on a beach.' ] from above image using my bottom up model.
If you have any problem please leave a comment at any time.

Thank you :)

**Comment by l3bjorge** [comment]
Thanks a lot for your answer!

This is my email: l3bjorge@gmail.com

I will switch to VinVL to see if I can get better captions and I will also try to run inference on the OSCAR+ finetuned model from their repository.

Thanks again! :)
